management/sql/client.go

```diff
@@ -99,6 +103,27 @@ func (c SqlDatabaseClient) CreateDatabase(server string, params DatabaseCreatePa
 	return err
 }
 
+// WaitForDatabaseCreation is a helper method which waits
+// for the creation of the database on the given server.
+func (c SqlDatabaseClient) WaitForDatabaseCreation(
+	server, database string,
+	cancel chan struct{}) error {
+	for {
+		select {
+		case <-cancel:
+			return management.ErrOperationCancelled
+		case <-time.After(management.DefaultOperationPollInterval):
```

Comment:
Now you introduced a delay (30sec) even though maybe the operation was already completed ok the first try :-) Please try to stick to the reference implementation I sent in this case.